pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,158 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	grpc_recovery "github.com/grpc-ecosystem/go-grpc-middleware/recovery"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	panicHandlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	t.Run("With_HTTP_Panic_Recovery", func(t *testing.T) {
+		handler := HTTPPanicRecoveryHandler(panicHandlerFunc, logger.MustNewLogger("text", "info", "unix"))
+
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		resp := httptest.NewRecorder()
+		require.NotPanics(t, func() {
+			handler.ServeHTTP(resp, req)
+		})
+
+		require.Equal(t, http.StatusInternalServerError, resp.Code)
+	})
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
```

Comment:
```suggestion
func TestUnaryPanicInterceptor(t *testing.T) {
	listener := bufconn.Listen(1024 * 1024)
	t.Cleanup(func() {
		listener.Close()
		goleak.VerifyNone(t)
	})
```

Reply:
goleak is failing: 
 recovery_test.go:107: found unexpected goroutines:
        [Goroutine 28 in state select, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x140002b4a30, {0x10127d5b8, 0x1400004c320})
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:76 +0xc8
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 24
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:52 +0x118
         Goroutine 26 in state select, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x140002b49c0, {0x10127d5b8, 0x1400004c280})
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:76 +0xc8
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 24
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:52 +0x118
         Goroutine 27 in state select, with google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run on top of the stack:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x140002b49f0, {0x10127d5b8, 0x1400004c2d0})
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:76 +0xc8
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 24
        	/Users/steven.meadows/go/pkg/mod/google.golang.org/grpc@v1.63.2/internal/grpcsync/callback_serializer.go:52 +0x118
        ]

Reply:
@smeadows-okta I added the line `goleak.VerifyNone(t)` and it works fine. Can you try again? Maybe ordering was off before?

Reply:
Done